File: osmdroid-android/src/main/java/org/osmdroid/views/drawing/MapSnapshotHandler.java
package org.osmdroid.views.drawing;

import android.os.Handler;
import android.os.Message;

import org.osmdroid.tileprovider.MapTileProviderBase;

public class MapSnapshotHandler extends Handler {

    private MapSnapshot mMapSnapshot;

    public MapSnapshotHandler(final MapSnapshot pMapSnapshot) {
        super();
        mMapSnapshot = pMapSnapshot;
    }

    @Override
    public void handleMessage(final Message msg) {
        if (msg.what == MapTileProviderBase.MAPTILE_SUCCESS_ID) {
            final MapSnapshot mapSnapshot = mMapSnapshot;
            if (mapSnapshot != null) { // in case it was destroyed just before
                mapSnapshot.refreshASAP();
            }
        }
    }

    public void destroy() {
        mMapSnapshot = null;
    }
}

File: osmdroid-android/src/main/java/org/osmdroid/views/drawing/MapSnapshotHandler.kt
package org.osmdroid.views.drawing

import android.os.Handler
import android.os.Message
import org.osmdroid.tileprovider.MapTileProviderBase

class MapSnapshotHandler(private var mMapSnapshot: MapSnapshot?) : Handler() {
    override fun handleMessage(msg: Message) {
        if (msg.what == MapTileProviderBase.Companion.MAPTILE_SUCCESS_ID) {
            val mapSnapshot = mMapSnapshot
            if (mapSnapshot != null) { // in case it was destroyed just before
                mapSnapshot.refreshASAP()
            }
        }
    }

    fun destroy() {
        mMapSnapshot = null
    }
}
